modules/rosa-sts-cluster-terraform.adoc

```diff
@@ -0,0 +1,516 @@
+variable "cluster_name" {
+  type        = string
+  description = "The name of the ROSA cluster to create"
+  validation {
+    condition     = can(regex("^[a-z][-a-z0-9]{0,13}[a-z0-9]$", var.cluster_name))
+    error_message = "ROSA cluster name must be less than 16 characters, be lower case alphanumeric, with only hyphens."
+  }
+}
+
+variable "additional_tags" {
+  default = {
+    Terraform   = "true"
+  }
+  description = "Additional AWS resource tags"
+  type        = map(string)
+}
+
+variable "path" {
+  description = "(Optional) The arn path for the account/operator roles as well as their policies."
+  type        = string
+  default     = null
+}
+
+variable "multi_az" {
+  type        = bool
+  description = "Multi AZ Cluster for High Availability"
+}
+
+variable "machine_type" {
+  description = "The AWS instance type that used for the instances creation."
+  type        = string
+  default     = "m5.xlarge"
+}
+
+variable "worker_node_replicas" {
+  description = "Number of worker nodes to provision. Single zone clusters need at least 2 nodes, multizone clusters need at least 3 nodes"
+  type        = number
+}
+
+
+variable "autoscaling_enabled" {
+  description = "Enables autoscaling. This variable requires you to set a maximum and minimum replicas range using the 'max_replicas' and 'min_replicas' variables."
+  type        = string
+  default     = "false"
+}
+
+variable "min_replicas" {
+  description = "The minimum number of replicas for autoscaling."
+  type        = number
+  default     = null
+}
+
+variable "max_replicas" {
+  description = "The maximum number of replicas not exceeded by the autoscaling functionality."
+  type        = number
+  default     = null
+}
+
+variable "proxy" {
+  default     = null
+  description = "cluster-wide HTTP or HTTPS proxy settings"
+  type = object({
+    http_proxy              = string           # required  http proxy
+    https_proxy             = string           # required  https proxy
+    additional_trust_bundle = optional(string) # a string contains contains a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store.
+    no_proxy                = optional(string) # no proxy
+  })
+}
+
+#Private link cluster Info
+variable "enable_private_link" {
+  type        = bool
+  description = "This enables private link"
+  default     = false
+}
+
+#VPC Info
+variable "create_vpc" {
+  type        = bool
+  description = "Create custom VPC for ROSA cluster"
+}
+
+variable "vpc_name" {
+  type        = string
+  description = "VPC Name"
+  default     = null
+}
+
+variable "vpc_cidr_block" {
+  type        = string
+  description = "value of the CIDR block to use for the VPC"
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the private subnets"
+  default     = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+}
+
+variable "public_subnet_cidrs" {
```

Comment:
The code for this is not included in this example so this variable can be skipped